bgpd/bgp_route.c

```diff
@@ -4501,6 +4506,91 @@ static void bgp_soft_reconfig_table(struct peer *peer, afi_t afi, safi_t safi,
 				return;
 			}
 		}
+	}
+}
+
+int bgp_soft_reconfig_table_index_sequencer(struct thread *thread)
+{
+	struct soft_reconfig_table_attr *attr = THREAD_ARG(thread);
+	struct listnode *node, *nnode;
+
+	attr->thread = thread;
+	bgp_soft_reconfig_table_index(attr->peer, attr->afi, attr->safi, NULL,
+				      NULL, attr->min_idx, attr->max_idx);
+
+	listnode_delete(bm->list_soft_reconfig_table, attr);
+	XFREE(MTYPE_SOFT_RECONFIG_ATTR, attr);
+
+	for (ALL_LIST_ELEMENTS(bm->list_soft_reconfig_table, node, nnode,
+			       attr)) {
+		thread_add_timer_msec(bm->master,
+				      bgp_soft_reconfig_table_index_sequencer,
+				      attr, 100, &attr->thread);
+		break;
+	}
+	return 0;
+}
+
+static void bgp_soft_reconfig_table(struct peer *peer, afi_t afi, safi_t safi,
+				    struct bgp_table *table,
+				    struct prefix_rd *prd)
+{
+	uint32_t idx, p_idx;
+	struct soft_reconfig_table_attr attr, *nattr;
+	bool init_threads;
+	unsigned long table_count;
+
+	if (table != NULL || prd != NULL) {
+		bgp_soft_reconfig_table_index(peer, afi, safi, table, prd, 0,
+					      0xffffffff);
+		return;
+	}
+	attr.peer = peer;
+	attr.afi = afi;
+	attr.safi = safi;
+	attr.thread = NULL;
+
+	init_threads = (bm->list_soft_reconfig_table->head == NULL);
```

Comment:
This seems odd.  If we are in the middle of a clear, we wouldn't schedule a new one?